server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -243,6 +245,34 @@ protected final DocsStats docsStats(IndexReader indexReader) {
         return new DocsStats(numDocs, numDeletedDocs, sizeInBytes);
     }
 
+    /**
+     * @throws AlreadyClosedException if the shard is closed
+     */
+    public ShardFieldStats shardFieldStats() {
+        try (var searcher = acquireSearcher("shard_field_stats", Engine.SearcherScope.INTERNAL)) {
+            return shardFieldStats(searcher.getLeafContexts());
+        }
+    }
+
+    protected ShardFieldStats shardFieldStats(List<LeafReaderContext> leaves) {
```

Comment:
I think that we should make this method static.

Reply:
Good point! I've changed it to be static cb2b70bfdd1529ce3bf023b9b150c04b20b30dde